hackathon_site/dashboard/frontend/src/components/general/ProductOverview/ProductOverview.test.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import SideSheetRight from "../SideSheetRight/SideSheetRight";
+import { ProductOverview, EnhancedCartForm } from "./ProductOverview";
+import {
+    render,
+    fireEvent,
+    waitForElementToBeRemoved,
+    waitFor,
+    waitForElement,
+} from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+describe("Product Overview", () => {
+    const testDetail = {
+        name: "Arduino",
+        type: "red",
+        total: 30,
+        available: 29,
+        img: "https://www.filterforge.com/more/help/images/size200.jpg",
+        category: ["MCU", "FPGA"],
+        manufacturer: "Canakit",
+        model_num: "Model 3B+",
+        datasheet: "link",
+        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
+        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
+        quantity: 3,
+    };
+
+    test("render", () => {
+        const addCartMock = jest.fn();
+
+        const { asFragment } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
+                <ProductOverview detail={testDetail} addCartFunction={addCartMock} />
+            </SideSheetRight>
+        );
+
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    test("add to cart button calls the correct function", () => {
+        const addCartMock = jest.fn();
+
+        const { getByText, getByRole } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
+                <ProductOverview detail={testDetail} addToCart={addCartMock} />
+            </SideSheetRight>
+        );
+
+        const button = getByText("ADD TO CART");
+
+        fireEvent.click(button);
+        expect(addCartMock).toHaveBeenCalledTimes(2);
```

Comment:
I think this is being caused by not using the Enhanced form in `ProductOverview`. 

However, you are also right about having `onSubmit` set for both the form and the submit button is redundant (and the Enhanced component may just be magically fixing it). It's up to you which you want to do; since the button has `type="submit"` the form's `onSubmit` handler will be called, so you can either remove the `onClick` from the button or the `onSubmit` from the form. Removing from the button probably makes the most sense.

Reply:
Actually, something I just remembered - when a form is submitted, it causes the page to reload. You can notice this at the moment by clicking the add to cart button.

This behaviour may be removed once you are using the Enhanced form in the drawer, if formik removes the default behaviour. If it doesn't, then we'll want to have the `onClick` on the button, and remove the `onSubmit` on the form (but I'm pretty sure formik does prevent the default form submit action).

Reply:
You're right formik removed the refreshing behavior